bcipy/language/model/mixture.py

```diff
@@ -0,0 +1,147 @@
+        """
+
+        if lm_params is not None:
+            if (lm_types is None) or (len(lm_types) != len(lm_params)):
+                raise InvalidModelException("Length of parameters does not match length of types")
+
+        if lm_weights is not None:
+            if (lm_types is None) or (len(lm_types) != len(lm_weights)):
+                raise InvalidModelException("Length of weights does not match length of types")
+            if not isclose(sum(lm_weights), 1.0, abs_tol=1e-05):
+                raise InvalidModelException("Weights do not sum to 1")
+
+        if lm_types is not None:
+            if lm_weights is None:
+                raise InvalidModelException("Model weights not provided")
+            if lm_params is None:
+                raise InvalidModelException("Model parameters not provided")
+            if not all(x in MixtureLanguageModel.supported_lm_types for x in lm_types):
+                raise InvalidModelException(f"Supported model types: {MixtureLanguageModel.supported_lm_types}")
+
+        super().__init__(response_type=response_type, symbol_set=symbol_set)
+        self.models = list()
+        self.response_type = response_type
+        self.symbol_set = symbol_set
+        self.lm_types = lm_types or self.parameters.get("mixture_types")
+        self.lm_weights = lm_weights or self.parameters.get("mixture_weights")
+        self.lm_params = lm_params or self.parameters.get("mixture_params")
+
+        self.load()
+
+    def supported_response_types(self) -> List[ResponseType]:
+        return [ResponseType.SYMBOL]
+
+    @staticmethod
+    def interpolate_language_models(lms: List[Dict[str, float]], coeffs: List[float]) -> List[Tuple]:
+        """
+        interpolate two or more language models
+        Args:
+            lms - output from the language models (a list of dicts with char as keys and prob as values)
+            coeffs - list of rescale coefficients, lms[0] will be scaled by coeffs[0] and so on
+        Response:
+            a list of (char, prob) tuples representing an interpolated language model
+        """
+        combined_lm = Counter()
+
+        for i, lm in enumerate(lms):
+            for char in lm:
+                combined_lm[char] += lm[char] * coeffs[i]
+
+        return list(sorted(combined_lm.items(), key=lambda item: item[1], reverse=True))
+
+    def predict(self, evidence: List[str]) -> List[Tuple]:
+        """
+        Given an evidence of typed string, predict the probability distribution of
+        the next symbol
+        Args:
+            evidence - a list of characters (typed by the user)
+        Response:
+            A list of symbols with probability
+        """
+
+        pred_list = list()
+
+        # Generate predictions from each component language model
+        for model in self.models:
+            pred = model.predict(evidence)
+            pred_list.append(dict(pred))
```

Comment:
You could use a list comprehension here:

    pred_list = [dict(model.predict(evidence)) for model in models]